RecoEgamma/EgammaIsolationAlgos/plugins/ParticleBasedIsoProducer.cc

```diff
@@ -184,3 +188,27 @@ void ParticleBasedIsoProducer::produce(edm::Event& theEvent, const edm::EventSet
   fillerElectrons.fill();
   theEvent.put(std::move(eleToPFCandIsoMap_p), valueMapElePFCandIso_);
 }
+
+void ParticleBasedIsoProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // particleBasedIsolation
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("valueMapEleToEG", "");
+  desc.add<std::string>("valueMapPhoToEG", "valMapPFEgammaCandToPhoton");
+  desc.add<edm::InputTag>("electronTmpProducer", edm::InputTag("gedGsfElectronsTmp"));
+  desc.add<edm::InputTag>("pfCandidates", edm::InputTag("particleFlow"));
+  desc.add<std::string>("valueMapElePFblockIso", "gedGsfElectrons");
+  desc.add<edm::InputTag>("electronProducer", edm::InputTag("gedGsfElectrons"));
+  desc.add<edm::InputTag>("photonTmpProducer", edm::InputTag("gedPhotonsTmp"));
+  desc.add<edm::InputTag>("pfEgammaCandidates", edm::InputTag("particleFlowEGamma"));
+  {
+    edm::ParameterSetDescription psd0;
+    psd0.add<std::string>("ComponentName", "pfBlockBasedIsolation");
+    psd0.add<double>("coneSize", 9999999999);
+    desc.add<edm::ParameterSetDescription>("pfBlockBasedIsolationSetUp", psd0);
+  }
+  desc.add<edm::InputTag>("photonProducer", edm::InputTag("gedPhotons"));
+  desc.add<std::string>("valueMapPhoPFblockIso", "gedPhotons");
+  descriptions.add("particleBasedIsolation", desc);
```

Comment:
just curious why you are renaming if from the default label `particleBasedIsoProducer -> particleBasedIsolation`?

Reply:
With the automatically generated `cfi` files, the plugin label is also the filename. Before, the filename was `particleBasedIsoProducer_cfi.py` and the label was particleBasedIsolation. I chose to stick with the same label, if that's okay.